dataset/src/main/scala/frameless/functions/Udf.scala

```diff
@@ -0,0 +1,75 @@
+package frameless
+package functions
+
+import org.apache.spark.sql.{Column, FramelessInternals}
+
+trait Udf {
+  def udf[T, A, R](
+     f: A => R)
+  (implicit
+     aenc: TypedEncoder[A],
+     reng: TypedEncoder[R]) =
+    (u: TypedColumn[T, A]) => {
```

Comment:
I see there was a confusion with my previous comment, I was proposing to make it 
```scala
  def udf[T, A, R](f: A => R)(
     implicit
     aenc: TypedEncoder[A],
     reng: TypedEncoder[R]
  ): (TypedColumn[T, A] => TypedColumn[T, R]) = { 
  u => ...
}
```

General principle is that public methods should always have type signatures, but we don't need to repeat it in body again.